src/rpc/mempool.cpp

```diff
@@ -910,6 +910,41 @@ static RPCHelpMan submitpackage()
     };
 }
 
+static RPCHelpMan getprioritisationmap()
+{
+    return RPCHelpMan{"getprioritisationmap",
+        "Returns a map of all user-created (see prioritisetransaction) fee deltas by txid, and whether the tx is present in mempool.",
+        {},
+        RPCResult{
+            RPCResult::Type::OBJ_DYN, "prioritisation-map", "prioritisation keyed by txid",
+            {
+                {RPCResult::Type::OBJ, "txid", "transaction txid", {
+                    {RPCResult::Type::STR_AMOUNT, "fee-delta", "transaction fee delta in " + CURRENCY_UNIT},
+                    {RPCResult::Type::BOOL, "in-mempool", "whether this transaction is currently in mempool"},
+                }}
+            },
+        },
+        RPCExamples{
+            HelpExampleCli("getprioritisationmap", "")
+            + HelpExampleRpc("getprioritisationmap", "")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+        {
+            NodeContext& node = EnsureAnyNodeContext(request.context);
+            CTxMemPool& mempool = EnsureMemPool(node);
+            UniValue rpc_result{UniValue::VOBJ};
+            const auto prioritisation_map{mempool.GetPrioritisationMap()};
+            for (const auto& [txid, delta_exists_pair] : prioritisation_map) {
+                UniValue result_inner{UniValue::VOBJ};
+                result_inner.pushKV("fee-delta", ValueFromAmount(delta_exists_pair.first));
+                result_inner.pushKV("in-mempool", delta_exists_pair.second);
```

Comment:
If you have txindex enabled, you could lookup the txid of any `in-mempool: false` transactions in the txindex here, and, if found, calculate how many confirmations the tx has, and report that. That seems like it would be fairly useful, though not sure it's actually worth the effort (compared to writing a script that just calls `getrawtransaction $txid 1 | jq .confirmations`).